src/Init/Notation.lean

```diff
@@ -687,6 +687,27 @@ syntax (name := checkSimp) "#check_simp " term "~>" term : command
 -/
 syntax (name := checkSimpFailure) "#check_simp " term "!~>" : command
 
+/--
+`#discr_tree_key  t` prints the discrimination tree keys for the type of a term `t` or an identifier.
+It uses the default configuration for generating keys.
+
+For example,
+```
+#discr_tree_key (∀ {a n : Nat}, bar a (OfNat.ofNat n) = default)
+-- @Eq Nat (bar _ (@OfNat.ofNat Nat _ _)) (@default Nat _)
```

Comment:
Just to drive down the point that `discr_tree_key` does not need an equality, maybe


```suggestion
#discr_tree_key (∀ {a n : Nat}, bar a (OfNat.ofNat n))
-- bar _ (@OfNat.ofNat Nat _ _)
```

Are the parentheses needed?

Reply:
They come from `Lean.Meta.DiscrTree.keysAsPattern` so that would need changing. Personally, I think they aid readability.